src/libraries/System.Runtime/ref/System.Runtime.cs

```diff
@@ -6530,7 +6530,8 @@ public partial class BinaryReader : System.IDisposable
         public virtual int Read(char[] buffer, int index, int count) { throw null; }
         public virtual int Read(System.Span<byte> buffer) { throw null; }
         public virtual int Read(System.Span<char> buffer) { throw null; }
-        protected internal int Read7BitEncodedInt() { throw null; }
+        public int Read7BitEncodedInt() { throw null; }
```

Comment:
Did we consider adding a public `Read7BitEncodedInt32` that just calls the existing `protected internal` one, in order to have a better public name?

Reply:
I have a feeling this came up in the review. The downside is that protected APIs are still "public" in that they're part of our API surface. So there would be the wart that you'd still have _both_ `Read7BitEncodedInt` and `Read7BitEncodedInt32`, both of which do the same thing. That would potentially cause more confusion than just sucking it up and sticking with only the slightly worse name.

Reply:
> The downside is that protected APIs are still "public" in that they're part of our API surface. So there would be the wart that you'd still have both Read7BitEncodedInt and Read7BitEncodedInt32, both of which do the same thing.

For code in types derived from BinaryReader... is that common?

Reply:
I went through the video again (https://youtu.be/nD5kZ2aXjdM?t=4467 is a direct link). The justification was that we'd still have doc issues to contend with, and it turns out there is a non-trivial number of applications which subclass the type anyway. The conclusion was that we should accept the naming wart rather than add a duplicate method to the API surface. We left the 32-bit returning property names unchanged when adding 64-bit properties to the `Process` class, for instance.

Reply:
Ok